### ISSUE TITLE
Fix the version logged by the REPL

### ISSUE THREAD

**Issue by microbit-matt-hillsdon** [microbit-foundation/micropython-microbit-v2-simulator#74]
Real 2.1.0 prints:

>MicroPython v1.18 on 2022-09-26; micro:bit v2.1.0 with nRF52833

Simulator updated to 2.1.0 prints:

>MicroPython 61c02c8-dirty on 2022-09-27; micro:bit v2.0.0 with nRF52833

I think the hash there is from the sim repo itself which isn't very helpful (yep, autolinked!). Ideally we'd provide confidence that the sim matched the device. 

Ideally the 2.0.0 would be taken from the micropython-microbit-v2 submodule and match it. Potentially we could look at adding a simulator version number too but perhaps not worthwhile.

Not looked into how this is generated yet.

**Comment by microbit-matt-hillsdon** [comment]
@dpgeorge I'd appreciate advice about the MicroPython version header generation. It looks like the code in src/Makefile is generating the correct version information but it's being overwritten by the subsequent MicroPython build. What's the best approach?

**Comment by dpgeorge** [comment]
Yeah I can see the problem, it's generating the correct mpversion.h but then that gets overwritten.  Not sure exactly how to solve this.

**Comment by dpgeorge** [comment]
This PR at upstream MicroPython will hopefully allow us to solve this issue here: https://github.com/micropython/micropython/pull/10044
